### PR TITLE
1241 accessibility testing

### PR DESCRIPTION
**Pull request by marcellmueller**
Implements #1241

rebased this on #1411 to fix the yarn cache issue so this could pass CI

For the tests themselves the initial plan was to write totally separate tests for accessibility, though I soon realized we were repeating so many steps that this seemed like the wrong approach. With this in mind I think we will have to be mindful to not repeat tests unnecessarily. I don't think we need to test the same areas with different roles and focus on making sure we test each page and test the forms in different states ie read only, with errors etc.

I've placed them after most of our Happo screenshots and I think we will have to consider if that scenario is tested before adding more screenshots for other roles. Testing each form page *might* seem excessive though I thought it was worth it to make sure all of our widgets were compliant.

We had some accessibility issues though it wasn't too bad:

- Our pages needed an `h1` header. A lot of our pages don't have an obvious, unique `h1` in the design so I've wrapped the title in the nav in one. There is some contention about using a non-unique page title as an h1 though given the design of our pages this seemed like the best option.
- The disabled form field text colour not having enough contrast. Fixed by #1296 - RJSF read only theme
- The red minus button in our Parent Operator array field doesn't have text in it which failed, so I added an `aria-label`.
- There was an error with the DataGrid pagination component. This one eluded me though I found an option for the DataGrid `experimentalFeatures={{ ariaV7: true }}` which fixed it. While looking into this I noticed that `mui-x/DataGrid` version 7 just came out and tried installing that but it had further accessibility errors that seemed unrelated to any customizations we added (I disabled a bunch of our custom cells and pagination). I will likely open a tech debt ticket before merging this to keep an eye on v7 and update it in the future.

### CODE REVIEW

**Comment by BCerki** [comment]
> Implements #1241
> 
> rebased this on #1411 to fix the yarn cache issue so this could pass CI
> 
> For the tests themselves the initial plan was to write totally separate tests for accessibility, though I soon realized we were repeating so many steps that this seemed like the wrong approach. With this in mind I think we will have to be mindful to not repeat tests unnecessarily. I don't think we need to test the same areas with different roles and focus on making sure we test each page and test the forms in different states ie read only, with errors etc.
> 
> I've placed them after most of our Happo screenshots and I think we will have to consider if that scenario is tested before adding more screenshots for other roles. Testing each form page _might_ seem excessive though I thought it was worth it to make sure all of our widgets were compliant.
> 
> We had some accessibility issues though it wasn't too bad:
> 
> * Our pages needed an `h1` header. A lot of our pages don't have an obvious, unique `h1` in the design so I've wrapped the title in the nav in one. There is some contention about using a non-unique page title as an h1 though given the design of our pages this seemed like the best option.
> * The disabled form field text colour not having enough contrast. Fixed by [817 rjsf read only theme #1296](https://github.com/bcgov/cas-registration/pull/1296) - RJSF read only theme
> * The red minus button in our Parent Operator array field doesn't have text in it which failed, so I added an `aria-label`.
> * There was an error with the DataGrid pagination component. This one eluded me though I found an option for the DataGrid `experimentalFeatures={{ ariaV7: true }}` which fixed it. While looking into this I noticed that `mui-x/DataGrid` version 7 just came out and tried installing that but it had further accessibility errors that seemed unrelated to any customizations we added (I disabled a bunch of our custom cells and pagination). I will likely open a tech debt ticket before merging this to keep an eye on v7 and update it in the future.

This all makes sense to me. I think testing each form page is a good idea; certain widgets could be non-compliant.

File: client/e2e/workflows/bceidbusiness/industry_user.spec.ts [hunk 1]
```diff
@@ -50,12 +54,16 @@ test.describe("Test Workflow industry_user", () => {
...
+    // 🔍 Assert the form is visible - needed to prevent analyzeAccessibility from failing
+    await selectOperatorPage.formIsVisible();
     // 📷 Cheese!
     pageContent = page.locator("html");
     await happoPlaywright.screenshot(page, pageContent, {
       component: "Select operator page",
       variant: "default",
     });
+    // ♿️ Analyze accessibility
+    await analyzeAccessibility(page);
```
Comment (BCerki) [inline client/e2e/workflows/bceidbusiness/industry_user.spec.ts:66]: It looks like every happo shot is accompanied by the accessibility test, so could consider making a helper to do both.

**Review by BCerki — APPROVED** [review]
Thanks for your work on this, yay accessibility testing! And awesome that you were able to fix the fails in the same PR.

Everything in here looks good (pending CI fixes), so I'm going to approve, but this PR only covers accessibility for the industry user roles. Cas role tests are getting happo tests in @shon-button's https://github.com/bcgov/cas-registration/pull/1337, so we can block the existing card on that, or make a new one for cas role accessibility testing.

File: client/e2e/workflows/bceidbusiness/industry_user.spec.ts (hunk 1 above)
Comment (marcellmueller) [reply client/e2e/workflows/bceidbusiness/industry_user.spec.ts:66]: Thanks, I have made a separate PR to simplify this and make sure I don't break our Happo screenshots here (still WIP): https://github.com/bcgov/cas-registration/pull/1426

**Pull request merged**